Document/0x04i-Testing-user-interaction.md

```diff
@@ -19,10 +19,46 @@ When you need personal information from a user for your business process, the us
+
+As a new requirement on both platforms, it's vital that these labels are accurate in order to provide user assurance and mitigate developer abuse.
+
 
 When additional data needs to be processed, you should ask the user for consent again. During that consent request it needs to be made clear how the user can revert from sharing the additional data. Similarly, when existing datasets of a user need to be linked, you should ask the user's consent about it.
 
+You can learn more about this and other privacy related topics here:
+
+- [iOS App Privacy Policy](https://developer.apple.com/documentation/healthkit/protecting_user_privacy#3705073)
+- [iOS Privacy Details Section on the App Store](https://developer.apple.com/app-store/app-privacy-details/)
+- [iOS Privacy Best Practices](https://developer.apple.com/documentation/uikit/protecting_the_user_s_privacy)
+- [Android App Privacy Policy](https://support.google.com/googleplay/android-developer/answer/9859455#privacy_policy)
+- [Android Data Safety Section on Google Play](https://support.google.com/googleplay/android-developer/answer/10787469)
+- [Android Privacy Best Practices](https://developer.android.com/privacy/best-practices)
+
+#### Testing Data Privacy
+
+You can use the following resources as a starting point for your analysis.
+
+**Android:**
+- [Review how the app collects and shares user data](https://developer.android.com/guide/topics/data/collect-share).
+- Verify if the app performs [Data Access Auditing](https://developer.android.com/guide/topics/data/audit-access) (available for Android 11 (API level 30) and higher) and list all the used attribution tags. You can use the [DataAccessAuditingKotlin sample app](https://github.com/android/permissions-samples/tree/master/DataAccessAuditingKotlin) as a reference.
+- Use the [Privacy Dashboard](https://developer.android.com/training/permissions/explaining-access#privacy-dashboard) from the Android settings (Android 12 (API level 31) and higher) to monitor app access to sensitive information.
+
+**iOS:**
+
+- Verify which [iOS Privacy Details](https://developer.apple.com/app-store/app-privacy-details/) does the app include on the App Store.
+- Verify if and how the app is using the [App Tracking Transparency Framework](https://developer.apple.com/documentation/apptrackingtransparency).
+- [Enable the App Privacy Report](https://developer.apple.com/documentation/network/privacy_management/inspecting_app_activity_data) from the iOS settings (iOS 15.2 and higher) to monitor app activity data. After using the app extensively, you can save the report as JSON file containing a collection of dictionaries of different types. Parse for the `type: "access"` to inspect all data access by category (camera, contacts, etc.) and the `type: "networkActivity"` to examine all network accesses.
+
+
+These are some examples of common violations that you should report:
```

Comment:
How would this work in real-life for the user and for us during verification? 
Would we verify that a one-time pop-up is explaining this (which will be anyway coming when accessing camera, contacts etc due to approving the permission), so the additional information is needed? Or would we simply verify the nutrition / data safety labels? 
Where would it be defined that something is sensitive data, like `An app that records a user’s screen and doesn't treat this data as personal or sensitive data subject to this policy.` This is defined by the developer in the app? What would be the testing process, the tester would run app activity `https://developer.apple.com/documentation/network/privacy_management/inspecting_app_activity_data` and then we check it against the privacy policy? 

I think it's quite a few options we are listing above, so we might want to clarify a bit further and give some more guidance, otherwise it might be a bit overwhelming what to do now exactly.

Reply:
Since we're testing MSTG-STORAGE-12:

> The app **educates the user** about the types of personally identifiable information processed, as well as security best practices the user should follow in using the app.


I think the testing strategy for now should be:
- look at **if and what** the app declares it collects/shares (in the App Store iOS Privacy Details and in Google Play Safety Section).
- check if the app is at least doing everything it could be doing to inform the user:
  - using prominent in-app disclosure of data access, collection, use, and sharing.
  - using the "App Tracking Transparency Framework" / "Data Access Auditing".

I'm going to reflect this on this sub-section (now there's more info than required). We should address that but as part of other test cases / requirements.